### PR TITLE
ci: add automated spell checking

### PR DESCRIPTION
**Pull request by dviererbe**
As discussed in the last MIR meeting. This PR adds the GitHub Action [`check-spelling`](https://github.com/marketplace/actions/check-spelling).

I already configured some terminology used in the canonical/ubuntu/MIR context.

Note: This spell checker is not perfect (e.g. normal words like `plugins`, `todo`, `architetcures` had to be added), but as a demonstration of its capabilities, it found these misspelled words #18.

### CODE REVIEW

**Comment by dviererbe** [comment]
You can see an example spell-checker error report here: https://github.com/dviererbe/ubuntu-mir/commit/9ae06d89d6dec06edaf3e69a54e7ceb6e86ad1c7#commitcomment-114607322

File: .github/actions/spelling/allow/linux.txt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+linux
+linuxfoundation
+local
+lock
+log
+mail
+media
+mnt
+opt
+opt
```
[the remark this answers is not in the document]
Comment (dviererbe) [reply .github/actions/spelling/allow/linux.txt:24]: Oh, I missed that. I will fix that

File: .github/actions/spelling/allow/ubuntu-misc.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+main
+mir
+motu
+motus
+multiverse
+nbs
+restricted
+sru
+triaged
+uife
```
Comment (slyon) [inline .github/actions/spelling/allow/ubuntu-misc.txt:18]: should this be "uefi"?

File: .github/actions/spelling/allow/formats.txt [hunk 3]
```diff
@@ -0,0 +1,70 @@
+ada
+asm
+asn
+assembly
+bash
+basic
+c
+c#
+c++
```
Comment (setharnold) [inline .github/actions/spelling/allow/formats.txt:9]: I thought the tool should have stripped out the punctuation here, were these needed? (same with `f#` later)

File: .github/actions/spelling/allow/formats.txt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+js
+json
+julia
+latex
+lisp
+lua
+markdown
+matlab
+object
+objective-c
```
Comment (setharnold) [inline .github/actions/spelling/allow/formats.txt:40]: oh yes, this too? is this needed?

**Review by setharnold — COMMENTED** [review]
Looks awesome, thanks :D

File: .github/actions/spelling/allow/formats.txt (hunk 3 above)
Comment (jsoref) [reply .github/actions/spelling/allow/formats.txt:9]: The default minimum word size is `3` characters and in 0.0.21 the only characters it considers are `a-z` and `'` (with some aliases for this character).

Comment (dviererbe) [reply .github/actions/spelling/allow/formats.txt]: Resolved with https://github.com/canonical/ubuntu-mir/pull/19/commits/d7312d2433518149bbec0295d625b2a67fb767e3

I added this to `.github/actions/spelling/allow/README.md` with https://github.com/canonical/ubuntu-mir/pull/19/commits/e8224fb04f45d8b89e5df77ba860b4b87658809b

File: .github/actions/spelling/allow/ubuntu-misc.txt (hunk 2 above)
Comment (dviererbe) [reply .github/actions/spelling/allow/ubuntu-misc.txt]: Resolved with https://github.com/canonical/ubuntu-mir/pull/19/commits/3e7dd8342c82d456296ba8538f2192b63f6df9ee

File: .github/actions/spelling/allow/linux.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dviererbe) [reply .github/actions/spelling/allow/linux.txt]: Resolved with https://github.com/canonical/ubuntu-mir/pull/19/commits/ec04ed11e4d943d887e625ea30624902003a01e2

File: .github/actions/spelling/allow/ubuntu-misc.txt (hunk 2 above)
Comment (dviererbe) [reply .github/actions/spelling/allow/ubuntu-misc.txt]: Thanks for catching this!

File: .github/actions/spelling/allow/formats.txt (hunk 3 above)
Comment (dviererbe) [reply .github/actions/spelling/allow/formats.txt]: Thanks for noticing that!

File: .github/actions/spelling/allow/ubuntu-misc.txt (hunk 2 above)
Comment (dviererbe) [reply .github/actions/spelling/allow/ubuntu-misc.txt]: Ohhh... I remember again that this was the acronym UI Freeze Exception 🤦🏻

**Comment by eslerm** [comment]
this looks great!

the tool finds [*illegal patterns*](https://github.com/dviererbe/ubuntu-mir/actions/runs/5057531705/jobs/9076410849#step:2:1) which may need to be resolved in a separate PR

**Comment by jsoref** [comment]
You really want to fix those things before/in this PR otherwise people who fork from once this merges will be confused/unhappy.

**Comment by dviererbe** [comment]
I updated the comments in `.github/actions/spelling/line_forbidden.patterns` to give users a quick action how to fix the error message.

I also added the error fixes to #18.

File: .github/actions/spelling/line_forbidden.patterns [hunk 5]
```diff
@@ -1,61 +1,53 @@
...
-# you might not want to check in code where you were debugging w/ `fit()`, in which case, you might want
-# to use this:
-#\bfit\(
-
-# s.b. GitHub
+# If you encounter this error, use spelling "GitHub" instead.
 \bGithub\b
 
-# s.b. GitLab
+# If you encounter this error, use spelling "GitLap" instead.
```
Comment (jsoref) [inline .github/actions/spelling/line_forbidden.patterns:4]:
```suggestion
# If you encounter this error, use spelling "GitLab" instead.
```

Comment (dviererbe) [reply .github/actions/spelling/line_forbidden.patterns]: I updated the last commit, thanks!

**Pull request merged**